Restructure the documentation to use jupyter book

tonyfast opened the pull request:
- [x] Restructures documentation to use jupyter book and markdown
- [x] Adds a deploy command to nox for a better developer experience
- [x] introduces and `ARCHITECTURE.md` file to describe that goal and layout of the repository.
- [x] introduces a jupyter accessibility statement
- [ ] deploy on github pages
- [x] deploy on readthedocs

tonyfast commented:
i def don't have the authority to push to gh pages. going to try for readthedocs next.

isabela-pf commented:
I think I have the ability to enable github pages on the repo if that's what you need?

tonyfast commented:
thanks @isabela-pf, i'm not sure the token i need to push to a branch. it doesn't matter really, i've got the docs working on readthedocs. 

https://jupyter-accessibility.readthedocs.io/en/restructure/README.html

gabalafou commented:
This is the current linear flow of the pages in the Jupyter book if you start on the home page and click the next button:

1. Project Jupyter Accessibility Working Group
2. jupyter accessibility statement
3. jupyter accessibility community information
4. General and jupyter specific accessibility resources
5. Team Guides
6. Mapping JupyterLab’s user interface to source code
7. Walkthrough Phosphor Accessibility PRs Meeting Notes (10.26.2020)
8. Hackathon resources
9. jupyter audits
10. jupyter accessibility funding
11. Developer References
12. Contributing to jupyter/accessibility
13. architecture document for the jupyter/accessibility repository

I will use the above numbers to make some questions and suggestions:

- what is the difference between 3 and 1?
- should 11 be blank?

- [ ] Fill in 2 because it's blank (maybe with a WIP sentence like this page is coming soon)
- [ ] Fill in 3 because it's blank 
- [ ] Move audit links from sidebar/toc and into 9, the audits page (it's currently a one-sentence page)
- [ ] Please contact xxxxxx on 10
- [ ] Move 6 and 7 out of the flow and just as links into page 4. If that's not possible, maybe move the entire section to the end of the book?

tonyfast commented:
thanks for the feedback @gabalafou . i addressed your comments

* we still need a good contact for funding.
* the resources were moved to the end for the meantime.
* i shuffled the audit info to the accessibility statement which was a good idea. we still need a better a11y statement.

isabela-pf reviewed (commented):
I can only speak to the organization, but I think this make sense. I was able to navigate and find most things I was looking for. I appreciate that you grouped many of our resources to provide a top-level table of contents with less options so that it may be more approachable for people.

- [ ] The one thing I could not find was the JupyterLab accessibility meeting minutes. The PR says they haven't been removed from the repo, so this might be my error. Are they not meant to render outside of github?
- [ ] Could you add alt text to the logo? I think `Jupyter Accessibility — Home` will work.

gabalafou commented on ARCHITECTURE.md:
```diff
@@ -0,0 +1,29 @@
+
+1. the `ja11y` python module
+2. `jupyterlab` node packages
+3. accessibility testing toolsP
```
```suggestion
3. accessibility testing tools
```

gabalafou commented on noxfile.py:
```diff
@@ -1,7 +1,47 @@
-    session.run("doit", *session.posargs)
+TOC = Path("docs", "_toc.yml")
+CONFIG = Path("docs", "_config.yml")
+CONF = Path("docs", "conf.py")
```
`CONF` unused?

tonyfast replied:
i used it now.

tonyfast commented on .github/workflows/ci.yml:
```diff
@@ -1,51 +1,51 @@
-          path: |
-            reports
-        if: always()
+# name: ci
```
i'd like to keep them for when we work out our ci tasks later.

tonyfast commented on .github/workflows/docs.yml:
```diff
@@ -0,0 +1,24 @@
+# name: jupyter book documentation
```
this one is close to working and i think we'll get to it with a focus on ci

tonyfast commented on .readthedocs.yaml:
```diff
@@ -0,0 +1,5 @@
+
+python:
+  install:
+  - requirements: docs/requirements.txt
```
it builds. we can think about linters in the future as the project progresses. we'll want to adhere to jupyters standard conventions whatever they may be.

tonyfast commented:
>     * [ ]  Could you add alt text to the logo? I think `Jupyter Accessibility — Home` will work.

@isabela-pf Nope. 😞  don't know how to at the moment. i see the dissonance, but i think we put that on a list of improvements we'll need to find a solution for.

i've resurfacing the meeting minutes

trallard commented on docs/statement.md:
```diff
@@ -0,0 +1,11 @@
+# `jupyter` accessibility statement
+
+As of Jan 31, 2022, `jupyterlab` nor `retrolab` meet WCAG 2.1 Level A compliance. 
```
Maybe worth adding - if you want to keep track of concrete tasks towards making these WCAG compliant check the roadmap and this and that board?

trallard commented:
>we still need a good contact for funding.

Funding for a11y efforts in general? - there is NUMFOCUS and an Open Collective - wether there is a way to fund a1yy work only this might be an item to discuss and might only be possible if we manage to consolidate a formal "a11y working group"

isabela-pf commented:
Merging so we can iterate on this in a more concrete form. Thanks @tonyfast!
